linked-list.rb

```diff
@@ -62,13 +62,35 @@ def display
        puts full_list.join("->")
    end
 
-   def include?(key)
+   def include?(value)
+     current = @head
+     while !(current.next_node == nil)
+       return true if current.value == value
+       current = current.next_node
+     end
+     return true if current.value == value
+     return false
    end
 
    def size
+     current = @head
+     size = 0
+     while !(current.next_node == nil)
+         size += 1
+         current = current.next_node
+     end
+     size += 1
    end
 
    def max
+     current = @head
+     max = current.value
+     while !(current.next_node == nil)
+       max = current.value if current.value > max
+       current = current.next_node
+     end
+     max = current.value if current.value > max
+    max
    end
```

Comment:
This looks great and works, but in all of these (include?, size, and max) you had to account for that last node separately -- if you change your loop to while current != nil instead of while current.next_node != nil then you wouldn't need that! I told you to model after display which does it the latter way -- because sometimes you need to do something different with the last case and I wanted that to be the pattern I showed -- but now I want to point out how we could have optimized the pattern just slightly. Hopefully that makes sense!